table/block_based/block_based_table_builder.cc

```diff
@@ -311,7 +312,10 @@ struct BlockBasedTableBuilder::Rep {
   // `kBuffered` state is allowed only as long as the buffering of uncompressed
   // data blocks (see `data_block_buffers`) does not exceed `buffer_limit`.
   uint64_t buffer_limit;
-
+  bool buffer_exceeds_global_memory_limit;
+  // TODO-t CacheRep is the refactored CacheRep class from WriteBufferManager
+  // TODO-t figure out a better name for cache_rep in the context of this class
+  CacheRep* cache_rep;
```

Comment:
Note[Implementation blocker]: 
we need to refactor/improve CacheRep from WriteBufferManager before we can progress from this line in this PR.